### PR TITLE
Added a project file structure with a running server

### PR DESCRIPTION
**Pull request by don336**

### CODE REVIEW

File: src/config/dev.env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+PORT = 6000
+DB_CONNECT = mongodb://localhost:27017/Heai-backend_DB
```
Comment (myrdstom) [inline src/config/dev.env]: create `.env.example` which should be pushed to GitHub. this file should be in the root directory. Add config folder to `gitignore`

**Comment by myrdstom** [comment]
Update the README.md to reflect changes with regards to running the server locally

**Comment by mcdavidlubega** [comment]
I suggest a few changes
- Create a config folder as well
- Also please correct the routes folder to be:  routes/api-v1/
- Keep folder naming structure lowercase

**Comment by myrdstom** [comment]
Setup `es6` with babel, not this `{type:"module"}` stuff

**Comment by mcdavidlubega** [comment]
Oh also please make a distinction between the dev dependencies and project dependencies.
For example, everything to do with babel should probably be a dev dependency.
```
 "dependencies": {
        "bcrypt": "^5.0.1",
        "celebrate": "^15.0.1",
        "env-cmd": "^10.1.0",
        "express": "^4.18.1",
        "joi": "^17.6.0",
        "jsonwebtoken": "^8.5.1",
        "mongoose": "^6.5.2"
    },
    "devDependencies": {
        "@babel/cli": "^7.18.10",
        "@babel/core": "^7.18.13",
        "@babel/node": "^7.18.10",
        "@babel/preset-env": "^7.18.10",
        "@babel/runtime": "^7.18.9",
        "nodemon": "^2.0.19",
    }
```

**Comment by don336** [comment]
> I suggest a few changes
> 
> * Create a config folder as well
> * Also please correct the routes folder to be:  routes/api-v1/
> * Keep folder naming structure lowercase

We actually do have a config folder, but it was ignored. I made .env.example file to illustrate what has to be done

**Review by mcdavidlubega — COMMENTED** [review]
This looks good in terms of project structure.

File: package.json [hunk 2]
```diff
@@ -16,5 +16,21 @@
   "bugs": {
     "url": "https://github.com/CohortOnePointOne/Heai-backend/issues"
   },
-  "homepage": "https://github.com/CohortOnePointOne/Heai-backend#readme"
+  "homepage": "https://github.com/CohortOnePointOne/Heai-backend#readme",
+  "dependencies": {
+    "body-parser": "^1.20.1",
+    "dotenv": "^16.0.3",
```
Comment (myrdstom) [inline package.json:22]: uninstall `dotenv`

**Comment by myrdstom** [comment]
@mcdavidlubega @don336  `config` folder should both be in the root and ignored. How to set the project up inclusive of creating a config folder should be explained in the `README.md`.

File: src/db/mongoose.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import mongoose from "mongoose";
+import dotenv from "dotenv"
```
Comment (myrdstom) [inline src/db/mongoose.js:2]: Remove all usages of `dotenv`

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import bodyParser from "body-parser";
+import express from "express";
+import dotenv from "dotenv"
```
Comment (myrdstom) [inline src/index.js]: same here

**Review by myrdstom — APPROVED** [review]

**Pull request merged**